src/github.com/getlantern/flashlight/util/http.go

```diff
@@ -2,21 +2,162 @@ package util
 
 import (
 	"crypto/tls"
+	"errors"
 	"fmt"
 	"net"
 	"net/http"
 	"net/url"
 	"time"
 
+	"github.com/getlantern/fronted"
 	"github.com/getlantern/golog"
 	"github.com/getlantern/keyman"
 	"github.com/getlantern/waitforserver"
 )
 
+const (
+	defaultAddr = "127.0.0.1:8787"
+)
+
 var (
 	log = golog.LoggerFor("flashlight.util")
+
+	// This is for doing direct domain fronting if necessary. We store this as
+	// an instance variable because it caches TLS session configs.
+	direct  = fronted.NewDirect()
+	timeout = time.After(60 * time.Second)
```

Comment:
`timeout` should be a variable instantiated at L98, or it will fire once and never again. Or, remove it at all and rely on underlie clients?